packages/@react-aria/overlays/src/useCloseOnScroll.ts

```diff
@@ -36,8 +36,8 @@ export function useCloseOnScroll(opts: CloseOnScrollOptions) {
 
     let onScroll = (e: MouseEvent) => {
       // Ignore if scrolling an scrollable region outside the trigger's tree.
-      let target = e.target as HTMLElement;
-      if (!triggerRef.current || !target.contains(triggerRef.current)) {
+      let target = e.target;
+      if (!triggerRef.current || !(target instanceof  HTMLElement) || !target.contains(triggerRef.current)) {
```

Comment:
Won't this prevent our overlays from automatically closing when the user scrolls the window?

Reply:
I don't think so. If target is an instance of `HTMLElement` then it behaves as before, and if it's not, it just prevents a runtime error which would be caused by accessing `.contains` On something that doesn't have it.
Technically, the check could be on `Element` rather than `HTMLElement` but in IE, it's defined only on HTMLElement instances, and I thought there ia no harm in checking on HTMLElement, even though IE is not supported by react-aria

Reply:
Perhaps 
```suggestion
      if (!triggerRef.current || ((target instanceof HTMLElement) && !target.contains(triggerRef.current))) {
```
since window.scroll is a [valid](https://developer.mozilla.org/en-US/docs/Web/API/Window/scroll). Thanks @snowystinger for pointing the above out

Reply:
You are right. The reason that's broken is that target can be `document` which is not an instance of `HTMLElement`. Actually, `.contains` is actually on `Node`'s prototype and document is an instance of `Node`. It's only in IE9 that `.contains` is on `HTMLElement`'s prototype. So I think The right change would be to check if the target is an instance of `Node`.

Reply:
Pushed some update that should fix both the issue in cypress and the regression. 
Note that while [the suggested change here](https://github.com/adobe/react-spectrum/pull/2341#discussion_r716918659) fix the regression, and also fixes the runtime error on cypress. It doesn't completely solve the issue on cypress. For some reason, the `screenshot` command in cypress triggers an artificial scroll event where the target is set to `window`. Checking with `instanceof Node` and bailing out otherwise would skip unwanted closure in cypress and works in other cases too.

Reply:
Sorry for the regression BTW, and thanks for catching it. I shouldn't have considered IE9 in the first place :P

Reply:
Haha, no worries, thanks for the fix!